lib/cors-anywhere.js

```diff
@@ -51,11 +59,18 @@ function isValidHostName(hostname) {
  * @param request {ServerRequest}
  */
 function withCORS(headers, request) {
-  headers['access-control-allow-origin'] = '*';
-  if (request.headers['access-control-request-method']) {
-    headers['access-control-allow-methods'] = request.headers['access-control-request-method'];
+
+  headers['access-control-allow-origin'] = request.corsOptions.wildcardOrigin ? '*' : request.headers['origin'];
```

Comment:
Meta-remark: If all browsers were bug-free, then I would not hesitate with sending back an explicit Origin+Vary header by default. Unfortunately this is not the case (e.g. https://crbug.com/426089) so we still need to support the wildcard.

Can you use an if-else block instead of a ternary operator (and include the Vary handling here, see also my review comments further down)?